pylint_venv.py

```diff
@@ -1,63 +1,116 @@
+active virtualenv.
 
 Activation logic taken from https://github.com/pypa/virtualenv
+
+Usage:
+
+- Add an init hook to your ``.pylintrc``::
+    init-hook=
+        try: import pylint_venv
+        except ImportError: pass
+        else: pylint_venv.inithook()
+
+- Add the init hook as command line argument::
+
+    pylint --init-hook="import pylint_venv; pylint_venv.inithook()"
+
+- Add the init hook as command line argument and explicitly pass a venv::
+
+    pylint --init-hook="import pylint_venv; pylint_venv.inithook('$(pwd)/env')"
+
 """
 
 import os
+import platform
 import site
 import sys
 
 
-def is_venv():
-    """Return true if a virtual environment is active."""
-    return getattr(sys, "base_prefix", sys.prefix) != sys.prefix or hasattr(
-        sys, "real_prefix"
-    )
+IS_WIN = platform.system() == "Windows"
+IS_PYPY = platform.python_implementation() == "PyPy"
+
 
+def is_venv():
+    """Return ``true`` if a virtual environment is active and Pylint is
+    installed in it.
 
-def activate_venv(active_site, venv=None):
     """
-    Search and activate a Virtual Environment.
+    is_conda_env = getattr(sys, "base_prefix", sys.prefix) != sys.prefix
+    is_virtualenv = hasattr(sys, "real_prefix")
+    return is_conda_env or is_virtualenv
+
+
+def detect_venv():
+    # Check for a virtualenv or Conda env
+    for var in ["VIRTUAL_ENV", "CONDA_PREFIX"]:
+        venv = os.getenv(var, "")
+        if venv:
+            return venv
+
+    # Check if a local ".venv" folder exists
+    cwd = os.getcwd()
+    exec_path = "Scripts" if IS_WIN else "bin"
+    if os.path.isfile(os.path.join(cwd, ".venv", exec_path, "activate")):
+        return os.path.join(cwd, ".venv")
+
+    return None
+
+
+def activate_venv(venv):
+    """Check for an active venv and add its paths to Pylint.
 
     Activate the virtual environment from:
-    - The `venv` param, if one is given.
-    - `VIRTUAL_ENV` environmental variable if set.
-    - a `.venv` folder in the current working directory
-    """
-    if venv is None:
-        venv = os.environ.get("VIRTUAL_ENV", None)
 
-    if venv is None:
-        cwd = os.getcwd()
-        exec_path = 'Scripts' if sys.platform == "win32" else 'bin'
-        if os.path.isfile(os.path.join(cwd, ".venv", exec_path, "activate")):
-            venv = os.path.join(cwd, ".venv")
-
-    if venv is not None:
-        os.environ["VIRTUAL_ENV"] = venv
-        os.environ["PATH"] = (
-            os.path.join(venv, "bin") + os.pathsep + os.environ.get("PATH", "")
-        )
-        base = venv
-        if sys.platform == "win32":
-            site_packages = os.path.join(base, "Lib", "site-packages")
+    - The *venv* param if one is given.
+    - ``VIRTUAL_ENV`` environmental variable if set.
+    - A ``.venv`` folder in the current working directory
+
+    """
+    if IS_PYPY:
+        site_packages = os.path.join(venv, "site-packages")
+    else:
+        if IS_WIN:
```

Comment:
I'd use an `elif` here to save an indentation level.

Reply:
Yes, that’s right. The original code had an addional `elif IS_JYTHON` which I removed …